File: translation_output/w_test/deepseekcoder/codenet/java/python/p00991.py
<｜begin▁of▁sentence｜>You are an expert Python programmer.Your task is to write a Python function to solve a programming problem.

### Instruction:
Translate the following java code to python and enclose your solution inside ```python```.
A sample test case is provided below:

Test input:
4 4 0 0 3 3

Expected output:
2


```
import java.util.*;
import java.math.*;
class p00991{
	public static void main(String[]Z){
		Scanner cin=new Scanner(System.in);
		int r=cin.nextInt(),c=cin.nextInt(),a1=cin.nextInt(),a2=cin.nextInt(),b1=cin.nextInt(),b2=cin.nextInt();
		int x=Math.abs(a1-b1),y=Math.abs(a2-b2);
		x=Math.min(x,r-x);y=Math.min(y,c-y);
		int z=1,C=Math.min(x,y);
		if(x==r-x)z*=2;
		if(y==c-y)z*=2;
		BigInteger b=BigInteger.ONE;
		for(int i=0;i<C;i++)b=b.multiply(new BigInteger(String.valueOf(x+y-i))).divide(new BigInteger(String.valueOf(i+1)));
		System.out.println(b.multiply(new BigInteger(String.valueOf(z))).mod(new BigInteger("100000007")));
	}
}
```

### Response:
```python
import sys
from math import *
from functools import reduce

def solve(r, c, a1, a2, b1, b2):
    x = abs(a1 - b1)
    y = abs(a2 - b2)
    x = min(x, r - x)
    y = min(y, c - y)
    z = 1
    C = min(x, y)
    if x == r - x:
        z *= 2
    if y == c - y:
        z *= 2
    b = reduce(lambda a, b: a * b, range(x + y - C + 1, x + y + 1)) // reduce(lambda a, b: a * b, range(1, C + 1))
    return b * z % 100000007

r, c, a1, a2, b1, b2 = map(int, sys.stdin.readline().split())
print(solve(r, c, a1, a2, b1, b2))
```

This Python code reads the input from the standard input, converts it to the appropriate types, and then calls the `solve` function with the converted inputs. The `solve` function calculates the result as per the Java code. The result is printed to the standard output.
<|EOT|>

File: translation_output/w_test/deepseekcoder/codenet/python/java/p00991.java
<｜begin▁of▁sentence｜>You are an expert Python programmer.Your task is to write a Python function to solve a programming problem.

### Instruction:
Translate the following python code to java and enclose your solution inside ```java```.
A sample test case is provided below:

Test input:
4 4 0 0 3 3

Expected output:
2


```
from math import factorial
def comb (x,y):
	return factorial(x)//factorial(x-y)//factorial(y)

w,h,ax,ay,bx,by=map(int,input().split())
dx=min(w-abs(ax-bx),abs(ax-bx))
dy=min(h-abs(ay-by),abs(ay-by))
ans=1
if dx*2==w:ans*=2
if dy*2==h:ans*=2
ans*=comb(dx+dy,dx)
print(ans%100000007)
```

### Response:
Here is the Java version of the provided Python code:

```java
import java.util.Scanner;

public class Main {
    private static final int MOD = 100000007;

    private static long factorial(int n) {
        long result = 1;
        for (int i = 2; i <= n; i++) {
            result = (result * i) % MOD;
        }
        return result;
    }

    private static long comb(int x, int y) {
        return (factorial(x) * modInv((int)factorial(x-y), MOD) % MOD * modInv((int)factorial(y), MOD) % MOD) % MOD;
    }

    private static long modInv(int a, int mod) {
        return power(a, mod-2, mod);
    }

    private static long power(long a, long b, long mod) {
        long result = 1;
        while (b > 0) {
            if ((b & 1) != 0) {
                result = (result * a) % mod;
            }
            a = (a * a) % mod;
            b >>= 1;
        }
        return result;
    }

    public static void main(String[] args) {
        Scanner scanner = new Scanner(System.in);
        int w = scanner.nextInt();
        int h = scanner.nextInt();
        int ax = scanner.nextInt();
        int ay = scanner.nextInt();
        int bx = scanner.nextInt();
        int by = scanner.nextInt();
        scanner.close();

        int dx = Math.min(w-Math.abs(ax-bx), Math.abs(ax-bx));
        int dy = Math.min(h-Math.abs(ay-by), Math.abs(ay-by));

        long ans = 1;
        if (dx*2 == w) ans = (ans * 2) % MOD;
        if (dy*2 == h) ans = (ans * 2) % MOD;
        ans = (ans * comb(dx+dy, dx)) % MOD;

        System.out.println(ans);
    }
}
```

This Java code does the same thing as the Python code. It reads the input, calculates the minimum distance `dx` and `dy`, and then calculates the answer using the combination formula. The answer is then printed out.
<|EOT|>
